lib/peatio/client/blockchain_client.rb

```diff
@@ -0,0 +1,133 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+module Peatio
+  module BlockchainClient
+    Error                  = Class.new(StandardError) # TODO: Rename to Exception.
+    ConnectionRefusedError = Class.new(StandardError) # TODO: Remove this.
+
+    class << self
+      #
+      # Returns API client for given blockchain key.
+      #
+      # @param key [String, Symbol]
+      #   The blockchain key.
+      # @return [BaseAPI]
+      def [](key)
+        blockchain = Blockchain.find_by(key: key)
+        if blockchain.try(:client).present?
+          "BlockchainClient::#{blockchain.client.capitalize}"
+        end.constantize.new(blockchain)
+      end
+    end
+
+    class Base
+      extend Memoist
+
+      #
+      # Returns the blockchain.
+      #
+      # @return [blockchain]
+      attr_reader :blockchain
+
+      def initialize(blockchain)
+        @blockchain = blockchain
+      end
+
+      #
+      # Returns hot wallet balance.
+      #
+      # @abstract Derived API clients must implement it.
+      #
+      # @param currency [String, Symbol]
+      # @return [BigDecimal]
+      def load_balance!(currency)
+        method_not_implemented
+      end
+
+      #
+      # Returns transaction details.
+      #
+      # TODO: Docs.
+      #
+      # @param txid [String]
+      # @return [Hash]
+      #   The deposit details.
+      def load_deposit!(txid)
+        method_not_implemented
+      end
+
+      #
+      # Created new address.
+      #
+      # TODO: Doc.
+      #
+      def create_address!(options = {})
+        method_not_implemented
+      end
+
+      #
+      # Creates new withdrawal and returns transaction ID.
+      #
+      # TODO: Doc.
+      #
+      def create_withdrawal!(issuer, recipient, amount, options = {})
+        method_not_implemented
+      end
+
+      # TODO: Doc.
+      def inspect_address!(address)
+        method_not_implemented
+      end
+
+      def convert_to_base_unit!(value)
```

Comment:
I think we need to make a fully Abstract Class here
The plugin api principle should be to implement Abstract class and include only peatio-core
@dinesh-skyach @ysv @ritesh-skyach

Reply:
@mod I believe we can convert the `Base` class to module and then we include the module in plugins. What do you say?